src/lib.rs

```diff
@@ -40,81 +60,162 @@ impl<T: AsRef<[u8]>> Display for HexFmt<T> {
 
 impl<T: AsRef<[u8]>> LowerHex for HexFmt<T> {
     fn fmt(&self, f: &mut Formatter) -> Result {
```

Comment:
`#[inline]`? Not 100% sure, but since this typcally called cross-crate, it should be worthwhile?